### PR TITLE
WINC-1292: Set up konflux bundle builds

### PR DESCRIPTION
**Pull request by sebsoto**
Changes default image to the WMCO controller image, as required by konflux. Adds seperate bundle Dockerfile for konflux builds.

### CODE REVIEW

File: bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -431,7 +431,7 @@ spec:
                       fieldPath: metadata.name
                 - name: OPERATOR_NAME
                   value: windows-machine-config-operator
-                image: REPLACE_IMAGE
```
Comment (jrvaldes) [inline bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml:434]: this change will break

https://github.com/openshift/windows-machine-config-operator/blob/8a4f95f905527ece51fa9fb2580764a7ff6645a4/hack/common.sh#L324

is there a workaround, or this hack feature will no longer work/supported?

File: bundle-konflux.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Workaround for https://issues.redhat.com/browse/STONEBLD-2496
```
Comment (jrvaldes) [inline bundle-konflux.Dockerfile:1]: does this has to be in the root? does it makes sense to place it in the /build dir along the others dockerfiles?

File: Makefile [hunk 3]
```diff
@@ -200,10 +197,17 @@ CATALOG_IMG ?= $(IMAGE_TAG_BASE)-catalog:v$(WMCO_VERSION) ifneq ($(origin CATALO
 catalog-build: opm
 	$(OPM) index add --container-tool podman --mode semver --tag $(CATALOG_IMG) --bundles $(BUNDLE_IMGS) $(FROM_INDEX_OPT)
 
+# Image URL to use all building/pushing image targets
+ifdef IMG
+REPLACE_IMG_CMD := $(KUSTOMIZE) edit set image quay.io/redhat-user-workloads/windows-machine-conf-tenant/windows-machine-config-operator/windows-machine-config-operator-master=$(IMG)
+else
+REPLACE_IMG_CMD := echo using default image
```
Comment (jrvaldes) [inline Makefile:204]: > echo using default image

is this statement executable?

**Review by jrvaldes — COMMENTED** [review]
@sebsoto  thanks for working on this. PTAL at my comments.

File: Makefile (hunk 3 above)
Comment (sebsoto) [reply Makefile:204]: Yep:
```
# output: 
using default image
/home/sebsoto/code/openshift/windows-machine-config-operator/bin/kustomize build config/manifests | operator-sdk generate bundle -q --overwrite --version 10.17.0 --channels="preview,stable" --default-channel="stable"
INFO[0000] Creating bundle.Dockerfile                   
INFO[0000] Creating bundle/metadata/annotations.yaml    
```

File: bundle-konflux.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# This stages should be removed when the issue is resolved
+FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_9_1.22 as builder
```
Comment (mansikulkarni96) [inline bundle-konflux.Dockerfile:3]: can we pull brew images in CI now?

Comment (sebsoto) [reply bundle-konflux.Dockerfile]: Not in CI, but I have given konflux a brew pull secret in order to workaround a lack of go 1.22 image in the "normal" build repositories.

Comment (sebsoto) [reply bundle-konflux.Dockerfile]: That was the recommended solution

**Comment by sebsoto** [comment]
/override ci/prow/aws-e2e-operator ci/prow/aws-e2e-upgrade ci/prow/azure-e2e-operator ci/prow/azure-e2e-upgrade ci/prow/gcp-e2e-operator ci/prow/nutanix-e2e-operator ci/prow/platform-none-vsphere-e2e-operator ci/prow/vsphere-proxy-e2e-operator ci/prow/vsphere-e2e-operator ci/prow/wicd-unit-vsphere

**Pull request merged**